src/main.ts

```diff
@@ -50,11 +50,15 @@ async function run(): Promise<undefined> {
       header
     )
 
-    if (!previous) {
+    if (!previous && !deleteOldComment) {
       await createComment(octokit, repo, pullRequestNumber, body, header)
       return
     }
 
+    if (!previous) {
+      return
+    }
+
     if (deleteOldComment) {
       await deleteComment(octokit, previous.id)
       return
```

Comment:
I see, a halt is intended for ["delete previous comment"](https://github.com/marocchino/sticky-pull-request-comment#delete-previous-comment) behavior

```suggestion
    if (deleteOldComment) {
      if (previous) {
        await deleteComment(octokit, previous.id)
      }
      return
    }

    if (!previous) {
      await createComment(octokit, repo, pullRequestNumber, body, header)
      return
```